### ISSUE TITLE
Crash in static_cluster.proxy-cluster test

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4371]
On `raft` c2f1010ba679baa09abad0d33245b453c52ec630:

```
error: Error in ./src/containers/scoped.hpp at line 93:
error: Assertion failed: [ptr_] 
error: Backtrace:
1: rethinkdb_backtrace(void**, int) at rethinkdb_backtrace.cc:100
2: backtrace_t::backtrace_t() at backtrace.cc:203
3: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:283
4: format_backtrace(bool) at backtrace.cc:198
5: report_fatal_error(char const*, int, char const*, ...) at errors.cc:83
6: scoped_ptr_t<real_table_persistence_interface_t>::get() const at scoped.hpp:93
7: do_serve(io_backender_t*, bool, base_path_t const&, metadata_file_t*, serve_info_t const&, os_signal_cond_t*) at serve.cc:282
8: serve_proxy(serve_info_t const&, os_signal_cond_t*) at serve.cc:539
9: run_rethinkdb_proxy(serve_info_t*, bool*) at command_line.cc:878
10: _ZNSt5_BindIFPFvP12serve_info_tPbES1_S2_EE6__callIvJEJLi0ELi1EEEET_OSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE+0x76 at functional:1145
11: _ZNSt5_BindIFPFvP12serve_info_tPbES1_S2_EEclIJEvEET0_DpOT_+0x26 at functional:1203
12: std::_Function_handler<void (), std::_Bind<void (*(serve_info_t*, bool*))(serve_info_t*, bool*)> >::_M_invoke(std::_Any_data const&) at functional:1779
13: std::function<void ()>::operator()() const at functional:2160
14: starter_t::run_wrapper(std::function<void ()> const&) at runtime.cc:61
15: std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)>::operator()(starter_t*, std::function<void ()> const&) const at functional:550
16: _ZNSt5_BindIFSt7_Mem_fnIM9starter_tFvRKSt8functionIFvvEEEEPS1_S4_EE6__callIvJEJLi0ELi1EEEET_OSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE+0x70 at functional:1145
17: _ZNSt5_BindIFSt7_Mem_fnIM9starter_tFvRKSt8functionIFvvEEEEPS1_S4_EEclIJEvEET0_DpOT_+0x26 at functional:1203
18: std::_Function_handler<void (), std::_Bind<std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)> (starter_t*, std::function<void ()>)> >::_M_invoke(std::_Any_data const&) at functional:1779
19: std::function<void ()>::operator()() const at functional:2160
20: callable_action_instance_t<std::function<void ()>&>::run_action() at callable_action.hpp:31
21: callable_action_wrapper_t::run() at runtime_utils.cc:43
22: coro_t::run() at coroutines.cc:205
23: coro_t* coro_t::spawn_sometime<std::function<void ()>&>(std::function<void ()>&&&) at coroutines.hpp:59
24: starter_t::on_thread_switch() at runtime.cc:57
25: linux_message_hub_t::on_event(int) at message_hub.cc:154
26: epoll_event_queue_t::run() at epoll.cc:115
27: linux_thread_pool_t::start_thread(void*) at thread_pool.cc:161
28: /lib/x86_64-linux-gnu/libpthread.so.0(+0x7e9a) [0x7fb9b5d6fe9a] at 0x7fb9b5d6fe9a (/lib/x86_64-linux-gnu/libpthread.so.0)
29: clone+0x6d at 0x7fb9b5a9c8bd (/lib/x86_64-linux-gnu/libc.so.6)
```

**Comment by VeXocide** [comment]
In CR 2971 by @danielmewes.

**Comment by VeXocide** [comment]
Merged to `raft` via commit 2b14fe7908565fba83c3754b63f75efc0999a8fb.
